### ISSUE TITLE
more the 8 daisy chained

### ISSUE THREAD

**Issue by klarson18** [wayoda/LedControl#15]
I would like to "daisychain" 10 7219 chips using ledcontrol.h but I am only successful with 8.  Is this a software issue or a chip issue?  Thanks

**Comment by voneiden** [comment]
As far as I know the packet size for the chip is 16 bytes, which has 8 bytes of data. That means a maximum of 7 no-op bytes and as such it is limited to a maximum of 8 chips.

https://datasheets.maximintegrated.com/en/ds/MAX7219-MAX7221.pdf

**Comment by klarson18** [comment]
Thank You for the response!  Makes sense.
Kyle Larson

**Comment by Sloaix** [comment]
![](http://www.cirmall.com/upload/images/Frt2MvD92tBGUFQT6J6aCvU0YZVr/1.png)

@voneiden

I do not think that's the reason.

**Comment by voneiden** [comment]
Yeah, I remember this was a wrong conclusion. More than 8 can be chained.

**Comment by akhileshghanshyam** [comment]
[include.docx](https://github.com/wayoda/LedControl/files/1708169/include.docx)
@klarson18 can u please send me the code to run the 8 matrix chain. because i have problem with my code     
i have tried my best to make it run 8 matrix chain instead of 4. it will be very kind of u. my email id ghanshyam6298@gmail.com
attached here is the code which i am using. this code is to chain 4 matrics. can you edit this code for me or if have any other code which can chain more than 4 matrices than please send me on above mentioned email id.

**Comment by akhileshghanshyam** [comment]
@voneiden can u please help me to chain 10 (8x8) matrices. I have code which is capable of running 4 matrices. i am trying to make it to run 10 matrices. but no succeeded yet. can you help me in any way. My email id ghanshyam6298@gmail.com   . my hardware part is done and is working but in group of 4+4+2. the attached here is the code i am using. can you make desired changes in this code to make it run 10 matrices. if not 10 than at least 8.
[include.docx](https://github.com/wayoda/LedControl/files/1708154/include.docx)

**Comment by DigaFabian** [comment]
@klarson18 It is hardcoded in the [LedControl.cpp](https://github.com/wayoda/LedControl/blob/master/src/LedControl.cpp#L50) file.

To get more than 8 matrices working go to your Arduino library folder, in my case it is located at:

> C:\Users\{Username}\Documents\Arduino\libraries\LedControl\src

There open the LedControl.cpp file with an editor like notepad++ and remove lines 50 and 51.
This way I was able to chain 15 matrices, though I am not sure what the maximum amount of matrices is.
